pallets/omnipool/src/mock.rs

```diff
@@ -0,0 +1,409 @@
+	pub static MIN_TRADE_AMOUNT: RefCell<Balance> = RefCell::new(Balance::default());
+}
+
+construct_runtime!(
+	pub enum Test where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+		Balances: pallet_balances::{Pallet, Call, Storage, Config<T>, Event<T>},
+		Omnipool: pallet_omnipool::{Pallet, Call, Storage, Event<T>},
+		Tokens: orml_tokens::{Pallet, Event<T>},
+	}
+);
+
+impl frame_system::Config for Test {
+	type BaseCallFilter = frame_support::traits::Everything;
+	type BlockWeights = ();
+	type BlockLength = ();
+	type Origin = Origin;
+	type Call = Call;
+	type Index = u64;
+	type BlockNumber = u64;
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type AccountId = u64;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	type Event = Event;
+	type BlockHashCount = ConstU64<250>;
+	type DbWeight = ();
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = pallet_balances::AccountData<Balance>;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type SS58Prefix = ();
+	type OnSetCode = ();
+	type MaxConsumers = ConstU32<16>;
+}
+
+impl pallet_balances::Config for Test {
+	type Balance = Balance;
+	type DustRemoval = ();
+	type Event = Event;
+	type ExistentialDeposit = ConstU128<1>;
+	type AccountStore = System;
+	type WeightInfo = ();
+	type MaxLocks = ();
+	type MaxReserves = ConstU32<50>;
+	type ReserveIdentifier = [u8; 8];
+}
+
+parameter_type_with_key! {
+	pub ExistentialDeposits: |_currency_id: AssetId| -> Balance {
+		0
+	};
+}
+
+impl orml_tokens::Config for Test {
+	type Event = Event;
+	type Balance = Balance;
+	type Amount = i128;
+	type CurrencyId = AssetId;
+	type WeightInfo = ();
+	type ExistentialDeposits = ExistentialDeposits;
+	type OnDust = ();
+	type MaxLocks = ();
+	type DustRemovalWhitelist = Everything;
+}
+
+struct WeightCap;
+impl Get<(u32, u32)> for WeightCap {
+	fn get() -> (u32, u32) {
+		ASSET_WEIGHT_CAP.with(|v| *v.borrow())
+	}
+}
+struct FeeAsset;
+impl Get<(u32, u32)> for FeeAsset {
+	fn get() -> (u32, u32) {
+		ASSET_FEE.with(|v| *v.borrow())
+	}
+}
+struct FeeProtocol;
+impl Get<(u32, u32)> for FeeProtocol {
+	fn get() -> (u32, u32) {
+		PROTOCOL_FEE.with(|v| *v.borrow())
+	}
+}
+struct MinLiquidity;
+impl Get<Balance> for MinLiquidity {
+	fn get() -> Balance {
+		MIN_ADDED_LIQUDIITY.with(|v| *v.borrow())
+	}
+}
+struct MinAmount;
+impl Get<Balance> for MinAmount {
+	fn get() -> Balance {
+		MIN_TRADE_AMOUNT.with(|v| *v.borrow())
+	}
+}
+
+parameter_types! {
+	pub const HDXAssetId: AssetId = HDX;
+	pub const LRNAAssetId: AssetId = LRNA;
+	pub const DAIAssetId: AssetId = DAI;
+	pub const PosiitionClassId: u32= 1000;
+
+	pub ProtocolFee: (u32,u32) = FeeProtocol::get();
+	pub AssetFee: (u32,u32) = FeeAsset::get();
+	pub AssetWeightCap: (u32,u32) = WeightCap::get();
```

Comment:
I think this is some duplication? You should be able to write 
```Rust
parameter_types! {
	pub AssetWeightCap: (u32,u32) = ASSET_WEIGHT_CAP.with(|v| *v.borrow());
}
```

Reply:
There are also storage parameter_types btw. (see [examples here](https://paritytech.github.io/substrate/master/frame_support/macro.parameter_types.html#examples))